lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -36,65 +41,120 @@ def get_problem_grade_distribution(course_id):
+        # Retrieve course object down to problems
+        course = modulestore().get_course(course_id, depth=4)
+
+        # Connect to analytics data client
+        client = Client(base_url=settings.ANALYTICS_SERVER_URL, auth_token=settings.ANALYTICS_API_KEY)
+
+        for section in course.get_children():
+            for subsection in section.get_children():
+                for unit in subsection.get_children():
+                    for child in unit.get_children():
+                        if child.location.category not in PROB_TYPE_LIST:
+                            continue
+
+                        problem_id = child.location
+                        problem = client.modules(course_id, problem_id)
+
+                        try:
+                            grade_distribution = problem.grade_distribution()
+                        except NotFoundError:
+                            grade_distribution = []
+
+                        for score in grade_distribution:
+                            total_student_count[problem_id] += score['count']
+
+                            if problem_id in prob_grade_distrib:
+                                if prob_grade_distrib[problem_id]['max_grade'] < score['max_grade']:
+                                    prob_grade_distrib[problem_id]['max_grade'] = score['max_grade']
+
+                                prob_grade_distrib[problem_id]['grade_distrib'].append((score['grade'], score['count']))
+                            else:
+                                prob_grade_distrib[problem_id] = {
+                                    'max_grade': score['max_grade'],
+                                    'grade_distrib': [(score['grade'], score['count']), ]
+                                }
 
     return prob_grade_distrib, total_student_count
 
 
-def get_sequential_open_distrib(course_id):
+def get_sequential_open_distrib(course_id, enrollment):
     """
     Returns the number of students that opened each subsection/sequential of the course
 
     `course_id` the course ID for the course interested in
 
+    `enrollment` the number of students enrolled in this course.
+
     Outputs a dict mapping the 'module_id' to the number of students that have opened that subsection/sequential.
     """
+    sequential_open_distrib = {}
 
-    # Aggregate query on studentmodule table for "opening a subsection" data
-    db_query = models.StudentModule.objects.filter(
-        course_id__exact=course_id,
-        module_type__exact="sequential",
-    ).values('module_state_key').annotate(count_sequential=Count('module_state_key'))
+    if enrollment <= settings.MAX_ENROLLEES_FOR_METRICS_USING_DB:
+        # Aggregate query on studentmodule table for "opening a subsection" data
+        queryset = models.StudentModule.objects.filter(
+            course_id__exact=course_id,
+            module_type__exact='sequential',
+        ).values('module_state_key').annotate(count_sequential=Count('module_state_key'))
```

Comment:
@stvstnfrd I think this is <a href="https://www.python.org/dev/peps/pep-0008/#indentation">the preferred style</a>, but I might not be understanding which parenthesis you're talking about.